packages/ethereum-group-pcd/src/EthereumGroupPCD.ts

```diff
@@ -0,0 +1,329 @@
+  };
+
+  pubkeyMembershipConfig = {
+    circuit:
+      "https://storage.googleapis.com/personae-proving-keys/membership/pubkey_membership.circuit",
+    witnessGenWasm:
+      "https://storage.googleapis.com/personae-proving-keys/membership/pubkey_membership.wasm",
+  };
+}
+
+export async function prove(
+  args: EthereumGroupPCDArgs
+): Promise<EthereumGroupPCD> {
+  if (args.identity.value === undefined) {
+    throw new Error(`missing argument identity`);
+  }
+
+  if (args.signatureOfIdentityCommitment.value === undefined) {
+    throw new Error(`missing argument signatureOfIdentityCommitment`);
+  }
+
+  if (args.merkleProof.value === undefined) {
+    throw new Error(`missing argument merkleProof`);
+  }
+
+  const deserializedIdentity = await SemaphoreIdentityPCDPackage.deserialize(
+    args.identity.value.pcd
+  );
+  const message = deserializedIdentity.claim.identity.commitment.toString();
+  const msgHash = ethers.utils.hashMessage(message);
+
+  const prover = new MembershipProver(pubkeyMembershipConfig);
+  await prover.initWasm();
+  const { proof, publicInput } = await prover.prove(
+    args.signatureOfIdentityCommitment.value,
+    Buffer.from(msgHash.slice(2), "hex"),
+    JSONBig({ useNativeBigInt: true }).parse(args.merkleProof.value)
+  );
+
+  const publicInputMsgHash = "0x" + publicInput.msgHash.toString("hex");
+
+  if (msgHash !== publicInputMsgHash) {
+    throw new Error(
+      `public input message hash ${publicInputMsgHash} does not match commitment ${message} hash ${msgHash} `
+    );
+  }
+
+  const semaphoreSignature = await SemaphoreSignaturePCDPackage.prove({
+    identity: {
+      argumentType: ArgumentTypeName.PCD,
+      pcdType: SemaphoreIdentityPCDTypeName,
+      value: args.identity.value,
+    },
+    signedMessage: {
+      argumentType: ArgumentTypeName.String,
+      value: Buffer.from(proof).toString("hex"),
+    },
+  });
+
+  return new EthereumGroupPCD(
+    uuid(),
+    {
+      publicInput: publicInput,
+      groupType:
+        args.groupType.value === "address"
+          ? GroupType.ADDRESS
+          : GroupType.PUBLICKEY,
+    },
+    {
+      signatureProof: await SemaphoreSignaturePCDPackage.serialize(
+        semaphoreSignature
+      ),
+      ethereumGroupProof: Buffer.from(proof).toString("hex"),
+    }
+  );
+}
+
+export async function verify(pcd: EthereumGroupPCD): Promise<boolean> {
+  const semaphoreSignature = await SemaphoreSignaturePCDPackage.deserialize(
+    pcd.proof.signatureProof.pcd
+  );
+  const proofValid = await SemaphoreSignaturePCDPackage.verify(
+    semaphoreSignature
+  );
+
+  // the semaphore signature of the group membership proof must be valid
+  if (!proofValid) {
+    return false;
+  }
+
+  // the string that the semaphore signature signed must equal to the ethereum
+  // signature of the commitment
+  if (semaphoreSignature.claim.signedMessage !== pcd.proof.ethereumGroupProof) {
+    return false;
+  }
```

Comment:
the comment here feels like it's saying something different from what the code is doing